policyenforcer.go

```diff
@@ -46,9 +46,33 @@ type ValidationReport struct {
 	ArtifactReports []*ValidationReport
 }
 
-// PolicyEnforcer is an interface with methods that make policy decisions.
+// Evaluator is an interface that defines methods to aggregate and evaluate
+// verification results generated for an artifact per validation request.
+type Evaluator interface {
+	// Pruned checks if the policy evaluation is completed for provided pair of
+	// subject and artifact. 
+	// It returns [ErrVerifierPruned] if the policy evaluation is completed for 
+	// the given verifier. 
+	// It returns [ErrArtifactPruned] if the policy evaluation is completed for 
+	// the given artifact regardless of verifiers.
+	Pruned(ctx context.Context, subjectDigest, artifactDigest, verifierName string) error
```

Comment:
It is not semantically to return `ErrVerifierPruned` or `ErrArtifactPruned` as errors and it is also error prone.